.dotnet/src/Custom/Common/InternalListHelpers.cs

```diff
@@ -0,0 +1,37 @@
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace OpenAI;
+
+internal static class InternalListHelpers
+{
+    internal delegate Task<ClientResult> AsyncListResponseFunc(string continuationToken, int? pageSize);
+    internal delegate ClientResult ListResponseFunc(string continuationToken, int? pageSize);
+
+    internal static AsyncPageableCollection<T> CreateAsyncPageable<T, U>(AsyncListResponseFunc listResponseFunc)
+        where U : IJsonModel<U>, IInternalListResponse<T>
+    {
+        async Task<ResultPage<T>> pageFunc(string continuationToken, int? pageSize)
+            => GetPageFromProtocol<T,U>(await listResponseFunc(continuationToken, pageSize).ConfigureAwait(false));
+        return PageableResultHelpers.Create((pageSize) => pageFunc(null, pageSize), pageFunc);
```

Comment:
I'm not 100% certain of this, but:
- Should the parameter to nextPageFunc also handle the page size?  
- Should it be possible to pass a non-null continuationToken into the firstPageFunc?

Those may all be smuggled in by the closure, but it might be helpful to add tests for the various cases just to validate that they all work.

Reply:
Thanks for looking at this even after the stealth-merge, Anne!

Yes, the closure does handle it -- I only realized this because Visual Studio grayed it out and provided the suggestion to "remove the unnecessary lambda." 😂

Since the signature matches, I believe this is implicitly the same as
`(continuationToken, pageSize) => pageFunc(continuationTokenSize)`
for the nextPageFunc.

It was just very convenient to able to plumb through the single closure (overloaded, in a sense) to avoid too much duplicated code.